mllib/src/main/scala/org/apache/spark/mllib/evaluation/BinaryClassificationMetrics.scala

```diff
@@ -43,10 +43,13 @@ import org.apache.spark.sql.DataFrame
 @Since("1.0.0")
 class BinaryClassificationMetrics @Since("1.3.0") (
     @Since("1.3.0") val scoreAndLabels: RDD[(Double, Double)],
-    @Since("1.3.0") val numBins: Int) extends Logging {
+    @Since("1.3.0") var numBins: Int) extends Logging {
```

Comment:
It can cause the count/confusion RDD to recompute. 
yes, you can set numBins when construct BinaryClassificationMetrics.  But when you want to change the numBins, if there is no setBins function, you have to construct another BinaryClassificationMetrics.